### PR TITLE
clarify specifics of webhook support

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/openshift-docs/issues/4290

@bparees @oatmealraisin ptal

### CODE REVIEW

**Comment by bparees** [comment]
Does the doc somewhere explain that the build that is run as a result of the hook is going to build exactly the commit/ref provided in the hook event?

**Comment by oatmealraisin** [comment]
@gabemontero lgtm 👍

**Comment by gabemontero** [comment]
@bparees @oatmealraisin  I've reworked to make the build's use of the webhook event payload more explicit per the request from @bparees  (I did not find where it was explicitly noted before).

Also, moved off of the use of NOTE's.

**Comment by oatmealraisin** [comment]
@gabemontero This also lgtm 👍

File: dev_guide/builds/triggering_builds.adoc [hunk 1]
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
 link:https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket],
 or Generic webhooks.
 
+{product-title} webhooks currently only supports their analogous versions of the push event for each of the Git based source code management
```
Comment (bparees) [inline dev_guide/builds/triggering_builds.adoc]: s/supports/support/

File: dev_guide/builds/triggering_builds.adoc [hunk 2]
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
...
+systems (`SCMs`).  All other event types are ignored.
+
+When the push events are processed, a confirmation is made as to whether the branch reference inside the event matches the branch reference in the corresponding
+`BuildConfig`.  If they match, then the exact commit reference noted in the webhook event is checked out for the {product-title} build.
```
Comment (bparees) [inline dev_guide/builds/triggering_builds.adoc]: "If they do not match, no build is triggered."

**Comment by gabemontero** [comment]
updates pushed

**Comment by bparees** [comment]
technically applies to earlier versions of OCP too but since the new hooks were introduced in 3.6 it's not worth the hassle of refactoring the doc for different versions.

File: dev_guide/builds/triggering_builds.adoc [hunk 3]
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
 link:https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket],
 or Generic webhooks.
 
+{product-title} webhooks currently only support their analogous versions of the push event for each of the Git based source code management
+systems (`SCMs`).  All other event types are ignored.
+
+When the push events are processed, a confirmation is made as to whether the branch reference inside the event matches the branch reference in the corresponding
+`BuildConfig`.  If they do not match, no build is triggered.
```
Comment (bparees) [inline dev_guide/builds/triggering_builds.adoc]: i think you lost the part where it said we will build the specific commit ref in the event?

Comment (gabemontero) [inline dev_guide/builds/triggering_builds.adoc]: I misinterpreted your prior comment then.  I'll revert and then attempt to reapply your last comment.

**Comment by gabemontero** [comment]
update pushed (re-do on interpreting that last comment)

File: dev_guide/builds/triggering_builds.adoc (hunk 3 above)
Comment (bparees) [inline dev_guide/builds/triggering_builds.adoc]: yes, sorry it was intended as an addendum, not a replacement... sloppy commenting on my part.

**Comment by bparees** [comment]
lgtm, @openshift/team-documentation ptal/merge

**Comment by vikram-redhat** [comment]
@bfallonf - PTAL.

File: dev_guide/builds/triggering_builds.adoc [hunk 4]
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
 link:https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket],
 or Generic webhooks.
 
+{product-title} webhooks currently only support their analogous versions of the push event for each of the Git based source code management
+systems (`SCMs`).  All other event types are ignored.
```
Comment (adellape) [inline dev_guide/builds/triggering_builds.adoc]: I don't think SCMs needs to be marked up here.

Comment (gabemontero) [inline dev_guide/builds/triggering_builds.adoc]: update pushed

File: dev_guide/builds/triggering_builds.adoc [hunk 5]
```diff
@@ -32,11 +32,18 @@ link:https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab],
 link:https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket],
 or Generic webhooks.
 
+{product-title} webhooks currently only support their analogous versions of the push event for each of the Git based source code management
+systems (SCMs).  All other event types are ignored.
+
+When the push events are processed, a confirmation is made as to whether the branch reference inside the event matches the branch reference in the corresponding
```
Comment (bfallonf) [inline dev_guide/builds/triggering_builds.adoc]: @gabemontero Is it OpenShift that confirms the above?

**Comment by bfallonf** [comment]
Thanks @gabemontero . I just have one question about the process then I can probably merge this.

File: dev_guide/builds/triggering_builds.adoc (hunk 5 above)
Comment (gabemontero) [inline dev_guide/builds/triggering_builds.adoc]: @bfallonf - yes, the confirmation is made within openshift, specifically the openshift master.

thanks

Comment (bfallonf) [inline dev_guide/builds/triggering_builds.adoc]: @gabemontero Thanks. I'll merge this and do a quick followup to include that.

**Pull request merged**